File: src/subscriber/scripts/subscriber_node.py
#!/usr/bin/env python

import rospy
from turtlesim.msg import Pose
def PoseCallback(msg):
 rospy.loginfo("I received position=(%4.2f, %4.2f), orientation(RAD)=%4.2f",msg.x, msg.y, msg.theta);

if __name__ == '__main__':
 sub = rospy.Subscriber('turtle1/pose', Pose, PoseCallback)
 rospy.init_node('subscriber_node')
 loop_rate = rospy.Rate(5)
 while not rospy.is_shutdown():
	 loop_rate.sleep()

File: src/subscriber/src/subscriber_node.cpp
#include <ros/ros.h>
#include <turtlesim/Pose.h>

void PoseCallback(const turtlesim::Pose& msg)
{
 ROS_INFO_STREAM("I received " << "position=(" << msg.x << ", " << msg.y << "), " << "orientation(RAD)=" << msg.theta);
}

int main(int argc, char ** argv)
{
 ros::init(argc, argv, "subscriber_node"); 

 ros::NodeHandle n; 
 
 ros::Subscriber sub = n.subscribe("turtle1/pose",1000,&PoseCallback); 
 
 ros::spin(); //  enters a loop pumping callback functions until CTRL + C

 return 0;
}
